src/Grid.js

```diff
@@ -11,14 +11,23 @@ export default class Grid {
         const newGrid = new Grid(this.width, this.height);
 
         for (let i = 0; i < this.height; i++) {
-            for (let j = 0; j < this.width  ; j++) {
+            for (let j = 0; j < this.width; j++) {
                 newGrid.grid[i][j].passable = this.grid[i][j].passable;
             }
         }
 
         return newGrid;
     }
 
+    obfuscateGrid(x, y, grid, visibleCells) {
+        for (let cell of visibleCells) {
+            grid.grid[cell.y][cell.x].terrain = cell.terrain;
+            grid.grid[cell.y][cell.x].passable = cell.passable;
```

Comment:
i feel like non-passable cells is really just another terrain type 🤔

Reply:
🐇